lib/simplifyExpression/arithmeticSearch/index.js

```diff
@@ -2,35 +2,40 @@ const ChangeTypes = require('../../ChangeTypes');
 const evaluate = require('../../util/evaluate');
 const Node = require('../../node');
 const TreeSearch = require('../../TreeSearch');
+const {build, query} = require('math-nodes');
 
 // Searches through the tree, prioritizing deeper nodes, and evaluates
 // arithmetic (e.g. 2+2 or 3*5*2) on an operation node if possible.
 // Returns a Node.Status object.
 const search = TreeSearch.postOrder(arithmetic);
 
+
+const isIntegerFraction = (node) => query.isNeg(node) ? isIntegerFraction(node.args[0]) : query.isDiv(node) && node.args.every(arg => Number.isInteger(query.getValue(arg)));
+
 // evaluates arithmetic (e.g. 2+2 or 3*5*2) on an operation node.
 // Returns a Node.Status object.
 function arithmetic(node) {
-  if (!Node.Type.isOperator(node)) {
+  if (!query.isOperation(node)) {
     return Node.Status.noChange(node);
   }
-  if (!node.args.every(child => Node.Type.isConstant(child, true))) {
+
+  if (!node.args.every(child => query.isNumber(child))) {
     return Node.Status.noChange(node);
   }
 
   // we want to eval each arg so unary minuses around constant nodes become
   // constant nodes with negative values
   node.args.forEach((arg, i) => {
-    node.args[i] = Node.Creator.constant(evaluate(arg));
+    node.args[i] = build.number(evaluate(arg));
   });
 
   // Only resolve division of integers if we get an integer result.
   // Note that a fraction of decimals will be divided out.
-  if (Node.Type.isIntegerFraction(node)) {
+  if (isIntegerFraction(node)) {
```

Comment:
yeah, put this function wherever the other Node.Type functions are coming from :) (I think Kevin already suggested that)